integration_tests/mockApis/curiousApi.ts

```diff
@@ -74,24 +73,30 @@ export default {
       },
     })
   },
-  stubGetCuriousGoals: (params: LearnerGoalsTestParams) => {
-    let jsonResp
-    if (params.emptyStates === false) {
-      jsonResp = LearnerGoalsMock
-    } else if (params.emptyStates === true) {
-      jsonResp = LearnerGoalsMockB
-    }
+  stubGetCuriousGoals: (options?: { prisonerNumber?: string; emptyStates?: boolean }) => {
```

Comment:
Using an `options` object, rather than the type `LearnerGoalsTestParams` (which has now been deleted)

The response body is now built from `aValidLearnerGoals` function call, where the goals are either the defaults, or are empty arrays.
(this is easier to grok than the previous version, where `LearnerGoalsMockB` was the version with empty arrays of Goals.